### PR TITLE
Configure docker for dapp console api

### PR DESCRIPTION
**Pull request by tremarkley**
Part of https://github.com/ethereum-optimism/ecopod/issues/870

Adds docker-compose for dapp-console-api.

### CODE REVIEW

**Comment by tremarkley** [comment]
* **#160** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/ecosystem/160?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @tremarkley and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: Dockerfile [hunk 1]
```diff
@@ -74,3 +74,25 @@ ENV NODE_EXTRA_CA_CERTS=/usr/local/share/ca-certificates/extra-ca-certificates.c
 EXPOSE 42069
 
 CMD [ "pnpm", "start" ]
+
+########################################
+# STEP 4: DAPP-CONSOLE-API
+########################################
+
+FROM node:18.19-bullseye-slim AS dapp-console-api
```
Comment (nitaliano) [inline Dockerfile:82]: we'll want this docker image to look more like what the paymaster is doing https://github.com/ethereum-optimism/ecosystem/blob/7c10c041d22e1acf6247ffc3836af2d979fc9191/Dockerfile#L38-L53. we want the base OS to be alpine and not bullseyes, the reason the event-log-indexer is doing it this way is ponder needed a slightly different setup with more deps

Comment (jakim929) [reply Dockerfile:82]: +1,

also one thing to note is the builder step builds **all** of the apps, so no need to rebuild, or refetch any dependencies

pnpm deploy the app then simply copy it in
```
RUN pnpm deploy --filter=dapp-console-api --prod /dapp-console-api

```

Comment (tremarkley) [reply Dockerfile]: thank you, good suggestions!

**Review by jakim929 — APPROVED** [review]

File: apps/dapp-console-api/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+      target: dapp-console-api
+    env_file: .env
+    environment:
+      - REDIS_URL=redis://redis:6379
+    healthcheck:
+      test: curl http://0.0.0.0:7300/api/v0/healthz
+    volumes:
+      - ../../certs/extra-ca-certificates.crt:/usr/local/share/ca-certificates/extra-ca-certificates.crt
+    ports:
+      - 7310:7300
```
Comment (jakim929) [inline apps/dapp-console-api/docker-compose.yml:30]: nit for later: wdyt about making this port something other than 7310 so it doesn't clash with paymaster-proxy

Comment (tremarkley) [reply apps/dapp-console-api/docker-compose.yml:30]: I can update that now!

**Pull request merged**